components/pedigree/resources/src/main/resources/pedigree/view/exportSelector.js

```diff
@@ -124,7 +132,43 @@ define([
             var exportType = $$('input:checked[type=radio][name="export-type"]')[0].value;
             //console.log("Import type: " + exportType);
 
-            if (exportType == "simpleJSON") {
+            if (exportType == "image") {
+                var imageType = $$('input:checked[type=radio][name="image-options"]')[0].value;
+
+                var _bbox = $('canvas').down().getBBox();
+                //set white background
+                var svgEl = $$("#canvas svg")[0];
+                var background = $$('.panning-background')[0];
+                svgEl.insertBefore(new Element('rect', {"id": "white-bbox-background",
+                                                        "width": _bbox.width,
+                                                        "height": _bbox.height,
+                                                        "x": _bbox.x,
+                                                        "y": _bbox.y,
+                                                        "fill": "white"}),
+                                   svgEl.firstChild
+                );
+
+                var svg = editor.getWorkspace().getSVGCopy();
+                var exportString = svg.getSVGText();
+                $('white-bbox-background').remove();
+
+                if (imageType == "png") {
+                    // generate PNG image on the back end
+                    var pedigreeImageExportServiceURL = editor.getExternalEndpoint().getPedigreeImageExportServiceURL();
+                    new Ajax.Request(pedigreeImageExportServiceURL, {
+                        method: 'POST',
+                        onSuccess: function(response) {
```

Comment:
yes, we did. But on the second thought @sdumitriu is right, I kind of though that if you click and it doesn't export you realize right away it doe snot work, but in theory users who are used to slow software may not realize it fails but assume it just takes forever.... so yeah, I guess need to tell the user